python_modules/dagster/dagster/_core/definitions/observable_asset.py

```diff
@@ -29,3 +38,45 @@ def an_asset() -> None:
         raise NotImplementedError()
 
     return an_asset
+
+
+# Our internal guts can handle empty strings for job name and run id
+# However making thse named constants for documentation, to encode where we are making the assumption,
+# and to allow us to change this more easily in the future, provided we are disciplined about
+# actually using this constants.
+RUNLESS_JOB_NAME = ""
+RUNLESS_RUN_ID = ""
+
+
+def report_runless_event(
+    instance: DagsterInstance, event_type: DagsterEventType, event_specific_data: EventSpecificData
+) -> None:
+    dagster_event = DagsterEvent(
+        event_type_value=event_type.value,
+        job_name=RUNLESS_JOB_NAME,
+        event_specific_data=event_specific_data,
+    )
+
+    instance.report_dagster_event(dagster_event=dagster_event, run_id=RUNLESS_RUN_ID)
+
+
+def report_runless_asset_materialization(
+    asset_materialization: AssetMaterialization,
+    instance: Optional[DagsterInstance] = None,
+) -> None:
+    report_runless_event(
+        instance or DagsterInstance.get(),
+        DagsterEventType.ASSET_MATERIALIZATION,
+        StepMaterializationData(asset_materialization),
+    )
+
+
+def report_runless_asset_observation(
+    asset_observation: AssetObservation,
+    instance: Optional[DagsterInstance] = None,
+) -> None:
+    report_runless_event(
+        instance or DagsterInstance.get(),
+        DagsterEventType.ASSET_OBSERVATION,
+        AssetObservationData(asset_observation),
+    )
```

Comment:
the style i ended up doing for reference 
https://github.com/dagster-io/dagster/pull/16571/files